include/irr/builtin/material_compiler/glsl/rasterization/impl.glsl

```diff
@@ -16,16 +18,20 @@ void instr_eval_execute(in instr_t instr, in MC_precomputed_t precomp, inout irr
 	irr_glsl_AnisotropicMicrofacetCache microfacet;
 	bsdf_data_t bsdf_data;
 
-	bxdf_eval_t bxdf_eval = bxdf_eval_t(0.0);
-
-	if (cosFactor > FLT_MIN)
+	if (!is_bxdf || positiveCosFactor)
 	{
-		//speculative execution
 		bsdf_data = fetchBSDFDataForInstr(instr);
 		ior = bsdf_data_decodeIoR(bsdf_data, op);
 		ior2 = matrixCompMult(ior, ior);
 		params = instr_getParameters(instr, bsdf_data);
-		const float ior_scalar = dot(CIE_XYZ_Luma_Y_coeffs, ior[0]);
+		microfacet = _microfacet;
```

Comment:
this should move to the second if statement... because COATING shouldn't need it (because it need not concern itself with the microfacet for specular reflection, i.e. the fresnel of the COATING part)